modules/ztp-configuring-a-static-ip.adoc

```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+//
+// *scalability_and_performance/ztp-zero-touch-provisioning.adoc
+
+[id="ztp-configuring-a-static-ip_{context}"]
+= Configuring static IP addresses for managed clusters
+
+Optionally, after creating the `AgentClusterInstall` custom resource, you can configure static IP addresses for the managed clusters.
+
+[NOTE]
+====
+You must create this custom resource before creating the `ClusterDeployment` custom resource.
+====
+
+.Prerequisites
+
+* Deploy and configure the `AgentClusterInstall` custom resource.
+
+.Procedure
+
+. Create a `NMStateConfig` custom resource:
++
+[source,yaml]
+----
+apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+ name: <cluster-name>
+ namespace: <cluster-name>
+ labels:
+   sno-cluster-<cluster-name>: <cluster-name>
+spec:
+ config:
+   interfaces:
+     - name: eth0
+       type: ethernet
+       state: up
+       mac-address: <mac-address> <1>
+       ipv4:
+         enabled: true
+         address:
+           - ip: <ip-address> <2>
+             prefix-length: <public-network-prefix> <3>
+         dhcp: false
+   dns-resolver:
+     config:
+       server:
+         - <dns-resolver> <4>
+   routes:
+     config:
+       - destination: 0.0.0.0/0
+         next-hop-address: <gateway> <5>
+         next-hop-interface: eth0
+         table-id: 254
+ interfaces:
+   - name: "eth0" <6>
+     macAddress: <mac-address> <7>
+----
+<1> `mac-address` is the MAC address of the target bare metal machine, that is, the same MAC address used in the `BareMetalHost` resource.
+<2> `ip-address` is the static IP address of the target bare metal machine.
+<3> `public-network-prefix` is the static IP address’s subnet for the target bare metal machine.
+<4> `dns-resolver` is the DNS server for the target bare metal machine.
+<5> `gateway` is the gateway for the target bare metal machine.
+<6> `name` must match the name specified in the `interfaces` section.
+<7> `mac-address` must match the MAC address specified in the `interfaces` section.
```

Comment:
Curiosity: is it standard RH practice/policy to omit the hyphen from the compound adjectives such as "bare-metal"?

Reply:
Usage is "bare metal". No hyphen.